sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/online/data_collector_schema.py

```diff
@@ -7,27 +7,36 @@
 import logging
 from typing import Any
 
-from marshmallow import fields, post_load
+from marshmallow import fields, post_load, validates, ValidationError
 
 from azure.ai.ml._schema import NestedField, PatchedSchemaMeta, StringTransformedEnum
 from azure.ai.ml._schema._deployment.online.destination_schema import DestinationSchema
 from azure.ai.ml._schema._deployment.online.request_logging_schema import RequestLoggingSchema
-from azure.ai.ml._schema._deployment.online.sampling_strategy_schema import SamplingStrategySchema
+from azure.ai.ml._schema._deployment.online.request_schema import RequestSchema
+from azure.ai.ml._schema._deployment.online.response_schema import ResponseSchema
+
 from azure.ai.ml.constants._common import RollingRate
 
 module_logger = logging.getLogger(__name__)
 
 
 class DataCollectorSchema(metaclass=PatchedSchemaMeta):
-    enabled = fields.Bool()
+    request = NestedField(RequestSchema)
+    response = NestedField(ResponseSchema)
     rolling_rate = StringTransformedEnum(
         required=False,
-        allowed_values=[RollingRate.HOUR, RollingRate.YEAR, RollingRate.MONTH, RollingRate.DAY, RollingRate.MINUTE],
+        allowed_values=[RollingRate.HOUR, RollingRate.DAY, RollingRate.MINUTE],
```

Comment:
super nit, I assume order doesn't matter but some kind of asc or desc order would be nice